git-open

```diff
@@ -57,6 +57,44 @@ if [[ -z "$giturl" ]]; then
   exit 1
 fi
 
+ssh_config=${ssh_config:-~/.ssh/config}
+# Resolves an ssh alias defined in ssh_config to it's corresponding hostname
+# echos out result, should be used within subshell $( ssh_resolve $host )
+# echos out nothing if alias could not be resolved
+ssh_resolve() {
+  domain="$1"
+  ssh_found=true
+  # Filter to only ssh_config lines that start with "Host" or "HostName"
+  resolved=$(while read -r ssh_line; do
+    # Split each line by spaces, of the form:
+    # Host alias [alias...]
+    # Host regex
+    # HostName resolved.domain.com
+    read -r -a ssh_array <<<"${ssh_line}"
+    ssh_optcode="${ssh_array[0]}"
+    if [[ ${ssh_optcode^^} == HOST ]]; then
+      # Host
+      ssh_found=false
+      # Iterate through aliases looking for a match
+      for ssh_index in $(seq 1 $((${#ssh_array[@]} - 1))); do
+        ssh_host=${ssh_array[$ssh_index]}
+        # shellcheck disable=SC2053
+        if [[ $domain == $ssh_host ]]; then
+          # Found a match, next HostName entry will be returned while matched
+          ssh_found=true
+          break
+        fi
+      done
+    elif $ssh_found && [[ ${ssh_optcode^^} == HOSTNAME ]]; then
+      # HostName, but only if ssh_found is true (the last Host entry matched)
+      # Replace all instances of %h with the Host alias
+      echo "${ssh_array[1]//%h/$domain}"
+    fi
+  done < <(grep -Pi "^\\s*Host(Name)?\\s+" "$ssh_config"))
```

Comment:
as I mentioned, i have a few repos where i use this ssh user pattern. (so thanks!)

however.. testing this out, i ran into the -P flag which isn't supported on my machine (typical mac w/ sierra). If i use `ggrep` (which i installed via homebrew coreutils), then this script works great.

Perhaps we can write the regex without requiring -P ?

Reply:
I was just using it for the `?` operator, does `grep -E` work on macOS?

Reply:
Perfect. Yup works great